deps/v8/src/objects.cc

```diff
@@ -18740,6 +18740,19 @@ Handle<ObjectHashTable> ObjectHashTable::Put(Handle<ObjectHashTable> table,
   if ((table->NumberOfDeletedElements() << 1) > table->NumberOfElements()) {
     table->Rehash(isolate->factory()->undefined_value());
   }
+  // If we're out of luck, we didn't get a GC recently, and so rehashing
+  // isn't enough to avoid a crash.
+  int nof = table->NumberOfElements() + 1;
+  if (!table->HasSufficientCapacity(nof)) {
```

Comment:
The call to HasSufficientCapacity() with n == NumberOfElements() + 1 doesn't look obviously correct to me.  Here is its implementation:
```c++
template <typename Derived, typename Shape, typename Key>
bool HashTable<Derived, Shape, Key>::HasSufficientCapacity(int n) {
  int capacity = Capacity();
  int nof = NumberOfElements() + n;
  int nod = NumberOfDeletedElements();
  // Return true if:
  //   50% is still free after adding n elements and
  //   at most 50% of the free elements are deleted elements.
  if (nod <= (capacity - nof) >> 1) {
    int needed_free = nof >> 1;
    if (nof + needed_free <= capacity) return true;
  }
  return false;
}
```
`capacity - nof` can be < 0 if `capacity < NumberOfElements() * 2 + 1` and right-shifting a negative number has implementation dependent behavior.  I think compilers all implement it as an arithmetic shift but I'd say it's better to avoid it altogether.

As well, because it uses `int`, there is a potential for integer overflow when NumberOfElements() is large.